wire-ios-sync-engine/Source/Synchronization/ZMOperationLoop+PushChannel.swift

```diff
@@ -17,11 +17,33 @@
 //
 
 import Foundation
+import WireAPI
 
 extension ZMOperationLoop: ZMPushChannelConsumer {
 
-    public func pushChannelDidReceive(_ data: ZMTransportData) {
-        if let events = ZMUpdateEvent.eventsArray(fromPushChannelData: data), !events.isEmpty {
+    public func pushChannelDidReceive(_ data: Data) {
+        // TODO: WPB-9612 remove event decoding monitoring.
+        // Since update event models aren't documented, we aren't yet completely sure
+        // that the new type-safe UpdateEvent model can successfully decode all event
+        // types. To test correctness, we try to decode every update event received
+        // through the push channel and simply log when it fails so we know how to
+        // fix it. Once we're sure it works, we should remove this.
+        do {
```

Comment:
probably we can put around a developer flag to avoid shipping this for future releases?

Reply:
Yeah I was thinking to do that, we'll need to inject some flags from the UI to the operation loop though.

Reply:
done